service/history/historyEngine.go

```diff
@@ -2458,6 +2458,11 @@ func (e *historyEngineImpl) ResetWorkflowExecution(
 	if err != nil {
 		return nil, err
 	}
+	if _, ok := baseMutableState.GetInFlightDecision(); !ok && baseMutableState.IsWorkflowExecutionRunning() {
+		return nil, &workflow.BadRequestError{
+			Message: "Cannot reset open workflow without completed decision task.",
```

Comment:
Is it accurate? -- A workflow can be open, but waiting for a timer to fire, so there is no in-flight decision, but it already have a completed decision. 

Also why we want to restrict this feature for only closed workflows?

Reply:
I think using `GetPendingDecision()` is more appropriate here. 

@longquanzheng  Basically we want to block the case when the decision task is not even scheduled when a reset is triggered as if workflow history only contains started event, the reset is kind of like a no-op.

Reply:
yeah I understand. But doesn't seem necessary to have that. There could be case that we need it to mitigate some server issues. So it would be nice to not strict that IMO.

Reply:
GetPendingDecision() also have the same issue. If a workflow is waiting for a timer to fire, there is no decision scheduled until the timer fires. The workflow has no pending decision nor started decision, but it doesn't mean that the workflow hasn't completed a decision before.

Reply:
The functions might have bad naming. From my understanding, getPending or getInFlight will return a decision task if there is completed one

Reply:
Changed to HasProcessedOrPendingDecision()